src/codegen/objects.rs

```diff
@@ -23,13 +23,17 @@ pub fn generate(env: &Env, root_path: &Path, mod_rs: &mut Vec<String>, traits: &
             }
         };
 
-        let path = root_path.join(file_name(&class_analysis.full_name));
+        let mod_name = obj.module_name.clone().unwrap_or_else(|| {
```

Comment:
May be better update `nameutils::file_name` to accept `&class_analysis.full_name` as `full_name` and `obj.module_name` as `configured_name` ?